packages/data/src/components/use-select/test/index.js

```diff
@@ -109,7 +122,7 @@ describe( 'useSelect', () => {
 		);
 
 		expect( selectSpyFoo ).toHaveBeenCalledTimes( 2 );
-		expect( selectSpyBar ).toHaveBeenCalledTimes( 2 );
+		expect( selectSpyBar ).toHaveBeenCalledTimes( 1 );
```

Comment:
So there's a perf improvement here or something?

Reply:
Kind of, because when you do a select like this:
```js
const name = useSelect( ( select ) => {
  return select( blockEditorStore ).getBlockName( clientId );
}, [ clientId ] );
```
then before this PR, when the `clientId` dependency changed, `useSelect` unsubscribed from the `blockEditorStore` and then immediately subscribed again. After this PR, this is no longer done, because `useSelect` knows that we're still selecting from this one store, that the set of observed stores didn't change. Only the `clientId` param has changed.

This kind of select is very common in block sidebar and toolbar, as they show info for the currently selected block.